File: LC/python/Evaluate Reverse Polish Notation.py
class Solution:
    # @param tokens, a list of string
    # @return an integer
    def evalRPN(self, tokens):
        stack = []
        for i in range(len(tokens)):
            if tokens[i]!='+' and tokens[i]!='-' and tokens[i]!='*' and tokens[i]!='/':
                stack.append(int(tokens[i]))
            else:
                num1 = stack.pop()
                num2 = stack.pop()
                if tokens[i] == '+':
                    tmp = num2+num1
                elif tokens[i] == '-':
                    tmp = num2-num1
                elif tokens[i] == '*':
                    tmp = num2*num1
                elif tokens[i] == '/':
                    if num2*num1<0:
                        tmp = -((-num2)/num1) #in python:-1/2 = -1
                    else:
                        tmp = num2/num1
                stack.append(tmp)
        return stack.pop()

File: LC/java/Evaluate Reverse Polish Notation.java
public class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> s = new Stack<Integer>();
        String op = "+-*/";
        
        
        for(String i:tokens){
            if(!op.contains(i)){
                s.push(Integer.valueOf(i));
            }
            else{
                if(s.size()<2){
                    return -1;
                }
                int a = s.pop();
                int b = s.pop();
                if(i.equals("+")){
                    s.push(a+b);
                }
                else if(i.equals("-")){
                    s.push(b-a);
                }
                else if(i.equals("*")){
                    s.push(a*b);
                }
                else{
                    s.push(b/a);
                }
            }
        }
        
        return s.pop();
    } 
}